src/lib/nftClaimer/utils.ts

```diff
@@ -0,0 +1,131 @@
+import { gql, ApolloClient, InMemoryCache, HttpLink } from '@apollo/client/core';
+import fetch from 'cross-fetch';
+import snapshot from '@snapshot-labs/snapshot.js';
+import { CID } from 'multiformats/cid';
+import { Wallet } from '@ethersproject/wallet';
+import { getAddress, isAddress } from '@ethersproject/address';
+import { BigNumber } from '@ethersproject/bignumber';
+import type { Proposal, Space } from '../../helpers/snapshot';
+
+const requiredEnvKeys = [
+  'NFT_CLAIMER_PRIVATE_KEY',
+  'NFT_CLAIMER_NETWORK',
+  'NFT_CLAIMER_DEPLOY_VERIFYING_CONTRACT',
+  'NFT_CLAIMER_DEPLOY_IMPLEMENTATION_ADDRESS',
+  'NFT_CLAIMER_DEPLOY_INITIALIZE_SELECTOR',
+  'NFT_CLAIMER_SUBGRAPH_URL'
+];
+
+const HUB_NETWORK = process.env.NETWORK || '1';
+
+const missingEnvKeys: string[] = [];
+requiredEnvKeys.forEach(key => {
+  if (!process.env[key]) {
+    missingEnvKeys.push(key);
+  }
+});
+
+if (missingEnvKeys.length > 0) {
+  throw new Error(
+    `NFT Claimer not configured properly, missing env keys: ${missingEnvKeys.join(', ')}`
+  );
+}
+
+export const signer = new Wallet(process.env.NFT_CLAIMER_PRIVATE_KEY as string);
+
+async function mintingAllowed(space: Space) {
+  return (await getSpaceCollection(space.id)).enabled;
+}
+
+export async function validateSpace(address: string, space: Space | null) {
+  if (!space) {
+    throw new Error('RECORD_NOT_FOUND');
+  }
+
+  if (
+    (process.env.NFT_CLAIMER_NETWORK !== '5' || process.env.NODE_ENV === 'test') &&
+    (await snapshot.utils.getSpaceController(space.id, HUB_NETWORK)) !== getAddress(address)
```

Comment:
Can we seperate these to variables, right now it is bit hard to understand, something like

```
const isTestEnvironment = process.env.NODE_ENV === 'test';
const isWrongNetwork = process.env.NFT_CLAIMER_NETWORK !== '5';
const isNotSpaceOwner = await snapshot.utils.getSpaceController(space.id, HUB_NETWORK) !== getAddress(address);

if ((isWrongNetwork || isTestEnvironment) && isNotSpaceOwner) {
  throw new Error('Address is not the space owner');
}
```

Reply:
Done some refactoring on this. Is not exactly like the suggestion, to avoid calling the spaceOwner function before checking the network